sourcemash/frontend/static/js/views/browse_view.js

```diff
@@ -33,23 +33,29 @@ Sourcemash.Views.BrowseView = Backbone.View.extend({
   },
 
   render: function() {
+    this.close();
+
     // Render parent view
-    this.$el.html(this.template({ models: this.collection.models }));
+    this.$el.html(this.template({ topics: Sourcemash.Views.BrowseView.FEED_TOPICS }));
     $('#typeahead').html(this.typeahead.render().el);
 
     // Render loading view
     this.loadingView = new Sourcemash.Views.LoadingView({loading: this.loading});
     this.$(".loading").html(this.loadingView.render().el);
 
-    // Render item cards
-    var feedCards = [];
-    this.collection.models.forEach(function(feed) {
-        var feedCardView = new Sourcemash.Views.FeedCardView({el: "#feed-card-" + feed.get('id'),
-                                                              model: feed});
-        feedCards.push(feedCardView)
-    });
+    feeds = this.collection;
+    Sourcemash.Views.BrowseView.FEED_TOPICS.forEach(function(topic) {
+      if (feeds.length > 1) {
```

Comment:
The render sometimes gets passed through twice, once with feeds = [] and another time with feeds full of stuff. This makes sure you don't try to start making feeds and views when there isn't anything to make.